temporal-sdk/src/test/java/io/temporal/workflow/SignalTest.java

```diff
@@ -0,0 +1,273 @@
+            .toBuilder()
+            .setWorkflowId(workflowId)
+            .build();
+    WorkflowTest.QueryableWorkflow client =
+        workflowClient.newWorkflowStub(WorkflowTest.QueryableWorkflow.class, options);
+
+    // SignalWithStart starts a workflow and delivers the signal to it.
+    BatchRequest batch = workflowClient.newSignalWithStartRequest();
+    batch.add(client::mySignal, "Hello ");
+    batch.add(client::execute);
+    WorkflowExecution execution = workflowClient.signalWithStart(batch);
+    testWorkflowRule.sleep(Duration.ofSeconds(1));
+
+    // Test client created using WorkflowExecution
+    WorkflowTest.QueryableWorkflow client2 =
+        workflowClient.newWorkflowStub(WorkflowTest.QueryableWorkflow.class, options);
+    // SignalWithStart delivers the signal to the already running workflow.
+    BatchRequest batch2 = workflowClient.newSignalWithStartRequest();
+    batch2.add(client2::mySignal, "World!");
+    batch2.add(client2::execute);
+    WorkflowExecution execution2 = workflowClient.signalWithStart(batch2);
+    assertEquals(execution, execution2);
+
+    testWorkflowRule.sleep(Duration.ofMillis(500));
+    assertEquals("World!", client2.getState());
+    assertEquals(
+        "Hello World!",
+        workflowClient.newUntypedWorkflowStub(execution, Optional.empty()).getResult(String.class));
+
+    // Check if that it starts closed workflow (AllowDuplicate is default IdReusePolicy)
+    WorkflowTest.QueryableWorkflow client3 =
+        workflowClient.newWorkflowStub(WorkflowTest.QueryableWorkflow.class, options);
+    BatchRequest batch3 = workflowClient.newSignalWithStartRequest();
+    batch3.add(client3::mySignal, "Hello ");
+    batch3.add(client3::execute);
+    WorkflowExecution execution3 = workflowClient.signalWithStart(batch3);
+    assertEquals(execution.getWorkflowId(), execution3.getWorkflowId());
+    client3.mySignal("World!");
+    WorkflowStub untyped = WorkflowStub.fromTyped(client3);
+    String result = untyped.getResult(String.class);
+    assertEquals("Hello World!", result);
+
+    // Make sure that cannot start if closed and RejectDuplicate policy
+    WorkflowTest.QueryableWorkflow client4 =
+        workflowClient.newWorkflowStub(
+            WorkflowTest.QueryableWorkflow.class,
+            options
+                .toBuilder()
+                .setWorkflowIdReusePolicy(
+                    WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE)
+                .build());
+    BatchRequest batch4 = workflowClient.newSignalWithStartRequest();
+    batch4.add(client4::mySignal, "Hello ");
+    batch4.add(client4::execute);
+    try {
+      workflowClient.signalWithStart(batch4);
+      fail("DuplicateWorkflowException expected");
+    } catch (WorkflowExecutionAlreadyStarted e) {
+      assertEquals(execution3.getRunId(), e.getExecution().getRunId());
+    }
+  }
+
+  @Test
+  public void testSignalUntyped() {
```

Comment:
This test fails when you use docker service configuration because SDK test rule always uses in-memory service and never calls `setUseExternalService`. You need to fix that bug, otherwise the test is looking fine.